homeassistant/components/met_eireann/weather.py

```diff
@@ -0,0 +1,186 @@
+"""Support for Met Éireann weather service."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import PLATFORM_SCHEMA, WeatherEntity
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    LENGTH_METERS,
+    LENGTH_MILES,
+    PRESSURE_HPA,
+    PRESSURE_INHG,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.pressure import convert as convert_pressure
+
+from .const import ATTRIBUTION, CONDITION_MAP, CONF_TRACK_HOME, DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Inclusive(
+            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.longitude,
+        vol.Optional(CONF_ELEVATION): int,
+    }
+)
+
+
+def format_condition(condition: str):
+    """Map the conditions provided by the weather API to those supported by the frontend."""
+    if condition is not None:
+        for key, value in CONDITION_MAP.items():
+            if condition in value:
+                return key
+    return condition
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Met Éireann weather platform."""
+    _LOGGER.warning("Loading Met Éireann via platform config is not supported")
```

Comment:
You should remove this deprecated setup.  If you intend to support yaml configuration, you have to do it from the integration level.